src/main/java/jenkins/plugins/office365connector/PotentialActionBuilder.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jenkins.plugins.office365connector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.plugins.office365connector.model.Fact;
+import jenkins.plugins.office365connector.model.PotentialAction;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.metadata.ContributorMetadataAction;
+import jenkins.scm.api.metadata.ObjectMetadataAction;
+import jenkins.scm.api.mixin.ChangeRequestSCMHead;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+/**
+ * @author Damian Szczepanik (damianszczepanik@github)
+ */
+public class PotentialActionBuilder {
+
+    private final Run run;
+
+    public PotentialActionBuilder(Run run) {
+        this.run = run;
+    }
+
+    public List<PotentialAction> buildPotentialAction(List<Fact> factsList) {
+        List<PotentialAction> potentialActions = pullRequestActionable(factsList);
+
+        potentialActions.add(buildViewBuild());
+        potentialActions.addAll(pullRequestActionable(factsList));
+
+        return potentialActions;
+    }
+
+    private PotentialAction buildViewBuild() {
```

Comment:
... and split into smaller method which will be easier to test later.

Reply:
Can we please rename this method 😞 buildActionable?

Reply:
This probably not as this is ``private`` and it produces one concrete action but the one above which is ``public`` for sure.

But then maybe rename this class to ``ActionableBuilder`` - what do you think?